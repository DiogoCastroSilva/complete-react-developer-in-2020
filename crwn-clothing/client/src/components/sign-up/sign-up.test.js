import React from 'react';
import { shallow } from 'enzyme';

import SignUp from './sign-up.component';


describe('testing SignUp component', () => {
    let wrapper;
    let onEmailSignInStart;
    let onGoogleSignInStart;

    beforeEach(() => {
        onEmailSignInStart = jest.fn();
        onGoogleSignInStart = jest.fn();

        const mockProps = {
            onEmailSignInStart: onEmailSignInStart,
            onGoogleSignInStart: onGoogleSignInStart
        };

        wrapper = shallow(<SignUp {...mockProps} />);
    });

    it('should render a SignUp component', () => {
        expect(wrapper).toMatchSnapshot();
    });
});