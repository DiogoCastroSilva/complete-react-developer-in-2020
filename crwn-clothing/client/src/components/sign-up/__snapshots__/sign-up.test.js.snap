// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing SignUp component should render a SignUp component 1`] = `
<styled.div>
  <styled.h2>
    I don't have an account!
  </styled.h2>
  <span>
    Sign up with your email and password
  </span>
  <form
    onSubmit={[Function]}
  >
    <FormInput
      handleChange={[Function]}
      label="Display Name"
      name="displayName"
      type="text"
      value=""
    />
    <FormInput
      handleChange={[Function]}
      label="Email"
      name="email"
      type="email"
      value=""
    />
    <FormInput
      handleChange={[Function]}
      label="Password"
      name="password"
      type="password"
      value=""
    />
    <FormInput
      handleChange={[Function]}
      label="Confirm Password"
      name="confirmPassword"
      type="password"
      value=""
    />
    <CustomButton
      type="submit"
    >
      SIGN UP
    </CustomButton>
  </form>
</styled.div>
`;
